feat: add non-destructive cache hydration support via isInitialHydrat…

zto-sbenning opened the pull request:
…ion context

- Add SetContext type with isInitialHydration flag
- Update Handler interface to accept optional SetContext in set()
- Implement NX behavior in redis-strings handler when isInitialHydration is true
- Update all handlers (local-lru, composite) to accept ctx parameter
- Propagate isInitialHydration flag from registerInitialCache to handlers
- Prevents runtime cache overwrites during app restarts and horizontal scaling

Addresses #23

AyronK commented on packages/nextjs-cache-handler/src/handlers/local-lru.ts:
```diff
@@ -99,7 +99,9 @@ export default function createHandler({
-    set(key, cacheHandlerValue) {
+    set(key, cacheHandlerValue, ctx) {
+      // LRU cache is in-memory and ephemeral, so NX behavior is not applicable
+      // We always set the value regardless of isInitialHydration flag
```
NH is a redis specific name, it shouldn't be mentioned in LRU implementation.

zto-sbenning commented on packages/nextjs-cache-handler/src/handlers/cache-handler.types.ts:
```diff
@@ -46,6 +46,20 @@ export type CacheHandlerParametersGetWithTags = [
+   * 
+   * @default false
+   */
+  isInitialHydration?: boolean;
```
Renamed `isInitialHydration` to `setOnlyIfNotExists` to better reflect the intent and avoid forcing a specific hydration strategy.

zto-sbenning commented on packages/nextjs-cache-handler/src/handlers/redis-strings.ts:
```diff
@@ -260,23 +260,27 @@ export default function createHandler({
+                }
+              : ctx?.isInitialHydration
+                ? { NX: true }
+                : undefined;
```
Applied this refactor.

zto-sbenning commented:
Thanks for the review!

I’ve addressed all the comments:
- made the non-overwriting behavior opt-in with default unchanged
- renamed the flag to `setOnlyIfNotExists` to better reflect intent
- aligned the LRU handler behavior without Redis-specific terminology
- refactored the Redis set options
- updated the README to document the initial cache write strategy

Please let me know if you’d like any further adjustments.
